### PR TITLE
[Admin] Extract a `ui/search_panel` component from `orders/cart`

### PR DESCRIPTION
**Pull request by elia**
## Summary

Extract the generic part from the `orders/cart` component making it reusable for the customer search.
Might not be the last iteration of the generalizing process, but so far this is what we need for the customer search, this PR is itself extracted and isolated for reviewing purposes from the work on the customer picker.

## Checklist

Check out our [PR guidelines](https://github.com/solidusio/.github/blob/master/CONTRIBUTING.md#pull-request-guidelines) for more details.

The following are mandatory for all PRs:

- [ ] I have written a thorough PR description.
- [ ] I have kept my commits small and atomic.
- [ ] [I have used clear, explanatory commit messages](https://github.com/solidusio/.github/blob/main/CONTRIBUTING.md#writing-good-commit-messages).

The following are not always needed:

- 📖 I have updated the README to account for my changes.
- 📑 I have documented new code [with YARD](https://www.rubydoc.info/gems/yard/file/docs/Tags.md).
- 🛣️ I have opened a PR to update the [guides](https://github.com/solidusio/edgeguides).
- ✅ I have added automated tests to cover my changes.
- 📸 I have attached screenshots to demo visual changes.

### CODE REVIEW

File: admin/spec/components/previews/solidus_admin/ui/search_panel/component_preview/overview.html.erb [hunk 1]
```diff
@@ -0,0 +1,24 @@
+<div class="mb-8">
+  <h6 class="text-gray-500 mb-3 mt-0">
+    Default
+  </h6>
+
+  <script>
+    window.addEventListener('<%= current_component.stimulus_id %>:submit', (event) => {
+      debugger
+      alert(`selected: ${event.detail.resultTarget.innerText}`)
+    })
```
Comment (rainerdema) [inline admin/spec/components/previews/solidus_admin/ui/search_panel/component_preview/overview.html.erb:10]: debugger 👆

File: admin/spec/components/previews/solidus_admin/ui/search_panel/component_preview/overview.html.erb [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    window.addEventListener('<%= current_component.stimulus_id %>:search', (event) => {
+      console.log('Search', event.detail)
```
Comment (rainerdema) [inline admin/spec/components/previews/solidus_admin/ui/search_panel/component_preview/overview.html.erb:12]: Is this console log necessary?

Comment (elia) [reply admin/spec/components/previews/solidus_admin/ui/search_panel/component_preview/overview.html.erb:12]: No, I thought it was ok to use it in lookbook to show off the provided data, I'll remove it

File: admin/spec/components/previews/solidus_admin/ui/search_panel/component_preview/overview.html.erb (hunk 1 above)
Comment (elia) [reply admin/spec/components/previews/solidus_admin/ui/search_panel/component_preview/overview.html.erb:10]: 🙏

**Review by rainerdema — APPROVED** [review]
👏

**Pull request merged**